nih-sparc/sparc-app
# Create new user flow

# Description

When user logs in they will be forcefully redirected to complete their profile which includes setting their: email, first name, and last name.

If they already have an existing pennsieve account then we will authenticate them and link their orcid account to their pennsieve account.

Once a user has completed their profile they can then view their profile info at the /user/profile page.

Once a user has completed their profile they cannot go through the flow again by navigating to the /welcome page

## Type of change


# How Has This Been Tested?

Locally.

In order to test a new user with no existing pennsieve account: 

1. Make sure to set your local environment variable SHOW_LOGIN_FEATURE=true
2. Register for a dev orcid account via: https://sandbox.orcid.org/register
3. Sign in via the header and attempt to navigate anywhere (you should be forced to the /welcome page)
4. Go through the prompts to follow and verify that when you return home you are logged in and can view the profile info
5. Verify you cannot enter the flow again by attempting to navigate to the /welcome page (you should be forcefully redirected back home)

In order to test a new user with an existing pennsieve account: 

1. Make sure to set your local environment variable SHOW_LOGIN_FEATURE=true
2. Register for a dev orcid account via: https://sandbox.orcid.org/register
3. Create a dev pennsieve account via: https://app.pennsieve.net/sign-up
4. Finish your pennsieve account sign up by following the confirmation email sent to you from pennsieve (you will be asked to create a password for your new pennsieve account)
5. In Sparc, sign in via the header and attempt to navigate anywhere (you should be forced to the /welcome page)
6. When prompted for your email enter the email that you used to create the pennsieve account (you should now be prompted for the pennsieve password you set to verify your identity)
7. Verify that the accounts get linked by signing back in using the same orcid login and viewing your profile info (The name and email should match that of your pennsieve account)
8. Verify you cannot enter the flow again by attempting to navigate to the /welcome page (you should be forcefully redirected back home)

# Checklist:

- [X] I have utilized components from the Design System Library where applicable
